File: LeetCode/Java/283-two-pointers-移动零.java
/*
 * @lc app=leetcode.cn id=283 lang=java
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
    public void moveZeroes(int[] nums) {
        int j = 0;
        for (int i : nums)
            if (i != 0)
                nums[j++] = i;
        for (; j < nums.length; j++)
            nums[j] = 0;
    }
}
// @lc code=end

File: LeetCode/C/283-two-pointers-移动零.cpp
/*
 * @lc app=leetcode.cn id=283 lang=cpp
 *
 * [283] 移动零
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    void moveZeroes(vector<int> &nums) {
        for (int a = 0, b = 0; a < nums.size(); a++)
            if (nums[a] != 0)
                swap(nums[a], nums[b++]);
    }
};
// @lc code=end
